### PR TITLE
Align logits and labels in OPT

### PR DESCRIPTION
**Pull request by MichelBartels**
# What does this PR do?
For other decoder models, the labels are shifted and the last logit of each sequence is removed so they align when computing the loss. This isn't done for OPT. This PR adds this feature.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@patrickvonplaten

### CODE REVIEW

**Comment by younesbelkada** [comment]
LGTM - this is because we took the model from BART which is an encoder decoder
wdyt @ArthurZucker @patrickvonplaten ?

**Review by ArthurZucker — APPROVED** [review]
Yes thanks for that ! Great initiative 👍🏻

**Review by younesbelkada — APPROVED** [review]

**Comment by patrickvonplaten** [comment]
Oh, good find @MichelBartels, we weren't super careful when writing the PR here - we should have aligned this with GPT2 right away. The problem is that, it's not really wrong to **not** shift the labels and people could have already written their training pipelines with OPT where the labels are shifted before being passed to the model. So this could be backwards breaking here. I however do think it's important to align OPT as much as possible with GPT2. 

@LysandreJik @sgugger - do you think we could fix this in a patch release?

**Comment by sgugger** [comment]
Yes it needs to be addressed ASAP to avoid breaking changes, a patch release today is fine by me.

**Comment by LysandreJik** [comment]
Yes, agreed! Let's merge this PR as-is, check if there are any other issues and do a patch release in a couple of hours.

**Review by LysandreJik — APPROVED** [review]
